### PR TITLE
Prompt to enable location before scanning.

### PR DESCRIPTION
**Pull request by MagneFire**
Having the location permission is not enough to successfully scan for watches.
Instead we need to ensure that locations are actually enabled too.

The logic is slightly changed so that Bluetooth is only enabled at app start (no need to scan when already paired).
If already paired, simply connect to the watch.
If not paired, prompt to enable location when not enabled, once enabled start scanning, else quit the app.

This solves https://github.com/AsteroidOS/AsteroidOSSync/issues/145.

### CODE REVIEW

**Comment by MagneFire** [comment]
Discovered a but! Please don't merge yet (unpairing causes a crash)

**Comment by MagneFire** [comment]
Last change has a small behavioral change such that the location message will be repeatedly shown again until the user has enabled locations.

So when location is disabled show dialog:
- User presses cancel -> app quits.
- User presses confirm -> open system location dialog
  - User enables location and returns -> App starts scanning
  - User returns without enabling location -> show dialog (So that the user can read the message if ignored the previous time(s) )

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -73,4 +73,6 @@
     <string name="OWM_API_KEY">OWM API Key</string>
     <string name="apply">Apply</string>
     <string name="cancel">Cancel</string>
+    <string name="location_disabled_message">In order to find watches the location needs to be enabled. Once paired you can safely disable it again.</string>
```
Comment (jrtberlin) [inline app/src/main/res/values/strings.xml:76]: I would rephrase it a tiny bit: "... the location service has to be enabled."
You are not enabling a location. You are enabling a services thats all about the location.
Other than that I think we should get that merged asap, wait for a few translations and then push out a new release.
